ic-factory/src/api.rs

```diff
@@ -26,6 +26,7 @@ pub trait FactoryCanister: Canister + Sized + PreUpdate {
     /// otherwise, cycles balances of `principal` is returned.
     /// If `principal` does not exists, `None` is returned.
     #[update(trait = true)]
+    #[allow(clippy::needless_lifetimes)]
     fn get_cycles<'a>(&'a self, principal: Option<Principal>) -> AsyncReturn<Option<Nat>> {
```

Comment:
I thought this was solved, the issue with the life times.
This should be valid to write as 
```rust
fn get_cycles(&self, principal: Option<Principal>) -> AsyncReturn<Option<Nat>>
```
and if `AsyncReturn` has a life time argument it should be written as `AsyncReturn<'_, Option<Nat>>`

Was this never fixed or do we have a regression?

Reply:
🤷

Reply:
Alright, leave it for now, we'll make a separate issue for this